cmip6_downscaling/config/config.py

```diff
@@ -1,104 +1,130 @@
-
-
-# Dask Executor  --------------------------------------------------------------------
-def return_dask_config():
-    dask_config_dict = {
-        "dask_executor_memory_limit": "16Gi",
-        "dask_executor_memory_request": "16Gi",
-        "dask_executor_threads_per_worker": 2,
-        "dask_executor_cpu_limit": 2,
-        "dask_executor_cpu_request": 2,
-        "dask_executor_adapt_min": 4,
-        "dask_executor_adapt_max": 20,
-    }
-    return dask_config_dict
-
-
-def return_extra_pip_packages():
-    extra_pip_packages = {
-        "EXTRA_PIP_PACKAGES": "git+https://github.com/carbonplan/cmip6-downscaling.git git+https://github.com/pangeo-data/scikit-downscale.git git+https://github.com/NCAR/xpersist.git"
-    }
-    return extra_pip_packages
-
-
-def return_env_config():
-    env_config = {
-        "AZURE_STORAGE_CONNECTION_STRING": return_azure_config()["connection_string"],
-        "EXTRA_PIP_PACKAGES": return_extra_pip_packages()["EXTRA_PIP_PACKAGES"],
-        'OMP_NUM_THREADS': '1',
-        'MPI_NUM_THREADS': '1',
-        'MKL_NUM_THREADS': '1',
-        'OPENBLAS_NUM_THREADS': '1',
-    }
-    return env_config
-
-
-def return_kubernetes_run_config():
-    kubernetes_run_config = KubernetesRun(
-        cpu_request=return_kubernetes_config()["kubernetes_cpu"],
-        memory_request=return_kubernetes_config()["kubernetes_memory"],
-        image=return_docker_config()["image"],
-        labels=return_prefect_config()["agent"],
-        env=return_env_config()["env_config"],
-    )
-    return kubernetes_run_config
-
-
-def return_dask_executor():
-
-    dask_executor = DaskExecutor(
-        cluster_class=lambda: KubeCluster(
-            make_pod_spec(
-                image=return_docker_config()["image"],
-                memory_limit=return_dask_config()["dask_executor_memory_limit"],
-                memory_request=return_dask_config()["dask_executor_memory_request"],
-                threads_per_worker=return_dask_config()["dask_executor_threads_per_worker"],
-                cpu_limit=return_dask_config()["dask_executor_cpu_limit]"],
-                cpu_request=return_dask_config()["dask_executor_cpu_request"],
-                env=return_env_config()["env_config"],
+class AbstractConfig:
+    # attributes
+    # prefect_storage
+    # prefect_run_config
+    # prefect_executor
+    # fsspec_fs
+    def __init__(self):
+        self.connection_string = os.environ.get("AZURE_STORAGE_CONNECTION_STRING")
+        self.agent = "az-eu-west"
+        self.storage = Azure("prefect")
+        self.extra_pip_packages = "git+https://github.com/carbonplan/cmip6-downscaling.git git+https://github.com/pangeo-data/scikit-downscale.git git+https://github.com/NCAR/xpersist.git"
+        self.intermediate_cache_path = "az://flow-outputs/intermediate"
+        self.results_cache_path = "az://flow-outputs/results"
+        self.serializer = "xarray.zarr"
+
+
+class CloudConfig(AbstractConfig):
+    def __init__(self):
```

Comment:
I think you want to decide on a semi-common set of parameters you may want to pass to each config (maybe memory, cpus, etc.). These should be configurable as keyword arguments to the class.